### PR TITLE
convert print() to debug(), #93

### LINKED ISSUE #93: debug()
child of #89

Calls the log function.
Almost same as print.

### PR DESCRIPTION
**Pull request by zippy**
This branch is about ready to go with converting `print()` to `debug()`.  It required a bit of reworking of a bunch of sections of the code to pass around the `context` (which contains the logger) so that it could properly be used.  I'm pretty sure we will need access to the context for other api commands too.

The one thing that's missing is that I couldn't figure out how to remove duplication TestLogger as used in the core_api crate because `use holochain_core::instance::tests::TestLogger` from the core_api crate fails.  So I stuck a copy of those functions in `test_utils` but note that you can't use those functions from inside the tests in the `holochain_core` crate because of weird naming issues with crates.  @sphinxc0re any idea how to fix this?

closes #93

### CODE REVIEW

File: agent/src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,9 @@ impl Agent {
             content: text.to_string(),
         })
     }
+    pub fn to_string(&self) -> String {
```
Comment (sphinxc0re) [inline agent/src/lib.rs]: Please insert an empty line above this one.

File: core/src/instance.rs [hunk 2]
```diff
@@ -218,17 +224,70 @@ pub fn dispatch_action(
...
+
+    impl Logger for TestLogger {
+        fn log(&mut self, msg: String) {
+            self.log.push(msg);
+        }
+    }
+
+    // trying to get a way to print out what has been logged for tests without a read function.
+    // this currently fails
+    impl fmt::Debug for TestLogger {
```
Comment (sphinxc0re) [inline core/src/instance.rs]: You could just `#[derive(Debug)]` this

Comment (zippy) [reply core/src/instance.rs]: done.

File: agent/src/lib.rs (hunk 1 above)
Comment (zippy) [reply agent/src/lib.rs]: done

**Comment by zippy** [comment]
in standup we decided to just add a ticket for the issue of cross-crate testing code #185 leave the duplicate code in place and get on with this ticket.

**Comment by thedavidmeister** [comment]
@zippy dumb question: why isn't the context in the state tree?

**Comment by zippy** [comment]
you know @thedavidmeister I started wondering that as I was coding this... I think the answer is that the context is meta stuff and also stuff that would be not be persisted.  For example, part of what's in the context is the Persister that persists the state.  No sure how you could persist the thing that persists you!
